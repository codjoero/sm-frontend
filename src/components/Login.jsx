import React from 'react';

const Login = () => (
    <form>
        <input type="text" id="email" placeholder="Email" />
        <br />
        <input type="text" id="password" placeholder="Password" />
        <br />
        <input type="submit" value="Login" />
    </form>
);

export default Login;
