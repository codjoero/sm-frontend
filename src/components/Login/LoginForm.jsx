import React from 'react';
import {
    Form, Grid, Segment, Button, Header, Message, Divider,
} from 'semantic-ui-react';
import { Link } from 'react-router-dom';
import '../../assets/startForms.css';


const LoginForm = props => (
    <div className="signForm">
        <Grid relaxed="very" stackable>
            <Segment color="green" className="regSegment">
                <Header as="h2" color="green" textAlign="center">
                    Login
                </Header>
                <Form error={props.invalidForm}>
                    <div className="spaceComponents">
                        {props.invalidForm
                            ? <Message error content={props.errorMsg} /> : null
                        }
                        <Form.Input
                            icon="user"
                            name="username"
                            iconPosition="left"
                            placeholder="Username"
                            onChange={props.inputChanged}
                            color="red"
                            required
                        />
                        <Form.Input
                            icon="lock"
                            name="password"
                            iconPosition="left"
                            type="password"
                            placeholder="Password"
                            onChange={props.inputChanged}
                            required
                        />
                        <Button
                            fluid
                            basic
                            disabled={props.disabled}
                            onClick={props.onSubmit}
                            loading={props.loading}
                            color="green"
                        >
                            Login
                        </Button>
                        <Divider horizontal>or</Divider>
                        <Link to="/register">
                            <Header as="h5" color="blue" textAlign="center">
                                Signup
                            </Header>
                        </Link>
                    </div>
                </Form>
            </Segment>
        </Grid>
    </div>
);

export default LoginForm;
