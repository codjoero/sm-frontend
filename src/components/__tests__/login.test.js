import React from 'react';
import { shallow } from 'enzyme';
import Login from '../Login';

const props = {};
describe('Login component', () => {
    it('should render without crashing', () => {
        const wrapper = shallow(<Login {...props} />);
        expect(wrapper).toMatchSnapshot();
    });
});
