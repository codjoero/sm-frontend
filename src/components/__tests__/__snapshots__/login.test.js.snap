// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login component should render without crashing 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Login />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <input
          id="email"
          placeholder="Email"
          type="text"
        />,
        <br />,
        <input
          id="password"
          placeholder="Password"
          type="text"
        />,
        <br />,
        <input
          type="submit"
          value="Login"
        />,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "id": "email",
          "placeholder": "Email",
          "type": "text",
        },
        "ref": null,
        "rendered": null,
        "type": "input",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": "br",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "id": "password",
          "placeholder": "Password",
          "type": "text",
        },
        "ref": null,
        "rendered": null,
        "type": "input",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": "br",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "type": "submit",
          "value": "Login",
        },
        "ref": null,
        "rendered": null,
        "type": "input",
      },
    ],
    "type": "form",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <input
            id="email"
            placeholder="Email"
            type="text"
          />,
          <br />,
          <input
            id="password"
            placeholder="Password"
            type="text"
          />,
          <br />,
          <input
            type="submit"
            value="Login"
          />,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "id": "email",
            "placeholder": "Email",
            "type": "text",
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": "br",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "id": "password",
            "placeholder": "Password",
            "type": "text",
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": "br",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "type": "submit",
            "value": "Login",
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
      ],
      "type": "form",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
